src/main/java/com/ifmo/epampractice/entities/UsersEntity.java

```diff
@@ -0,0 +1,200 @@
+        this.email = email;
+    }
+
+    public String getPasswordHash() {
+        return passwordHash;
+    }
+
+    public void setPasswordHash(String passwordHash) {
+        this.passwordHash = passwordHash;
+    }
+
+    public UserRole getRole() {
+        return role;
+    }
+
+    public void setRole(UserRole role) {
+        this.role = role;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public LocalDate getBirthDate() {
+        return birthDate;
+    }
+
+    public void setBirthDate(LocalDate birthDate) {
+        this.birthDate = birthDate;
+    }
+
+    public LocalDateTime getSignUpDate() {
+        return signUpDate;
+    }
+
+    public void setSignUpDate(LocalDateTime signUpDate) {
+        this.signUpDate = signUpDate;
+    }
+
+    public int getPassId() {
+        return passId;
+    }
+
+    public void setPassId(int passId) {
+        this.passId = passId;
+    }
+
+    public int getDrivingLicenseId() {
+        return drivingLicenseId;
+    }
+
+    public void setDrivingLicenseId(int drivingLicenseId) {
+        this.drivingLicenseId = drivingLicenseId;
+    }
+
+    public String getContactPhone() {
+        return contactPhone;
+    }
+
+    public void setContactPhone(String contactPhone) {
+        this.contactPhone = contactPhone;
+    }
+
+    public String getAddress() {
+        return address;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    public boolean isBlocked() {
+        return isBlocked;
+    }
+
+    public void setBlocked(boolean blocked) {
+        isBlocked = blocked;
+    }
+
+    public BigDecimal getReputation() {
+        return reputation;
+    }
+
+    public void setReputation(BigDecimal reputation) {
+        this.reputation = reputation;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        UsersEntity that = (UsersEntity) o;
+
+        if (id != that.id) return false;
+        if (passId != that.passId) return false;
+        if (drivingLicenseId != that.drivingLicenseId) return false;
+        if (isBlocked != that.isBlocked) return false;
+        if (!email.equals(that.email)) return false;
+        if (!passwordHash.equals(that.passwordHash)) return false;
+        if (role != that.role) return false;
+        if (!name.equals(that.name)) return false;
+        if (!birthDate.equals(that.birthDate)) return false;
+        if (!signUpDate.equals(that.signUpDate)) return false;
+        if (!contactPhone.equals(that.contactPhone)) return false;
+        if (!address.equals(that.address)) return false;
+        return reputation.equals(that.reputation);
+    }
+
+    @Override
+    public int hashCode() {
+        int result = id;
+        result = 31 * result + email.hashCode();
```

Comment:
what happens if email is null?

Reply:
fixed
even though email should never be null in the first place (NOT NULL check in DB)
maybe it would be better to throw exception if someone passes null email in constructor/setter?